doc/howto/cross_compiling/cross_compiling_for_android.md

```diff
@@ -0,0 +1,153 @@
+# Build PaddlePaddle for Android
+
+There are two approaches to build PaddlePaddle for Android:
+
+1. Use a Docker container, or
+2. cross-compiling on Linux.
+
+## Docker-based Cross-Compiling
+
+Docker-based cross-compiling is the recommended approach because Docker runs on all major operating systems, including Linux, Mac OS X, and Windows.
+
+### Build the Docker Image
+
+The following steps pack tools we need to build PaddlePaddle into a Docker image.
+
+```bash
+$ git clone https://github.com/PaddlePaddle/Paddle.git
+$ cd Paddle
+$ docker build -t paddle:dev-android . -f Dockerfile.android
+```
+
+### Build the Inference Library
+
+We can run this Docker image to build PaddlePaddle's inference library for Android:
+
+```bash
+$ docker run -it --rm -v $PWD:/paddle -e "ANDROID_ABI=armeabi-v7a" -e "ANDROID_API=21" paddle:dev-android
+```
+
+The Docker image accepts two arguments `ANDROID_ABI` and `ANDROID_API`:
+
+| Argument        | Optional Values         | Default |
+|-----------------|-------------------------|---------|
+|`ANDROID_ABI`    |`armeabi-v7a, arm64-v8a` | `armeabi-v7a` |
+|`ANDROID_API`    |`>= 21` | `21` |
+
+The ARM-64 architecture (`arm64-v8a`) requires at least level 21 of Android API.
+
+The default entry-point of the Docker image,[`paddle/scripts/docker/build_android.sh`](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/scripts/docker/build_android.sh) generates the [Android cross-compiling standalone toolchain](https://developer.android.com/ndk/guides/standalone_toolchain.html) according to the value of `ANDROID_ABI` and `ANDROID_API`.  For information about other configuration parameters, please go on reading.
+
+Above command line generates and outputs the inference library in `$PWD/install_android` and puts third-party libraries in `$PWD/install_android/third_party`.
+
+## Linux-based Cross-Compiling
+
+The Linux-base approach is to run steps in `Dockerfile.android` manually on a Linux x64 computer.
```

Comment:
Rephrase: The Linux-based approach to cross-compile is to run steps in `Dockerfile.android` manually on a Linux x64 computer.

Reply:
Done